### PR TITLE
Add unit test for utils

### PR DESCRIPTION
**Pull request by bandinib-amzn**
*Issue #, if available:*

*Description of changes:*

- Add unit test for utils
- Fix broken tests.
- updated snapshots.

```
opendistro_security % yarn test:jest_ui public/apps/configuration/utils**           
yarn run v1.22.4
$ node ./test/run_jest_tests.js --config ./test/jest.config.ui.js public/apps/configuration/utils
 PASS  public/apps/configuration/utils/delete-confirm-modal-utils.test.tsx
 PASS  public/apps/configuration/utils/display-utils.test.tsx
 PASS  public/apps/configuration/utils/form-row.test.tsx
 PASS  public/apps/configuration/utils/name-row.test.tsx
 PASS  public/apps/configuration/utils/password-edit-panel.test.tsx
 PASS  public/apps/configuration/utils/panel-with-header.test.tsx
 PASS  public/apps/configuration/utils/resource-validation-util.test.tsx
 PASS  public/apps/configuration/utils/tenant-utils.test.tsx
 PASS  public/apps/configuration/utils/toast-utils.test.tsx
 PASS  public/apps/configuration/utils/action-groups-utils.test.tsx
 PASS  public/apps/configuration/utils/role-list-utils.test.tsx
 PASS  public/apps/configuration/utils/index-permission-utils.test.tsx
 PASS  public/apps/configuration/utils/role-mapping-utils.test.tsx
 PASS  public/apps/configuration/utils/internal-user-list-utils.test.tsx
 PASS  public/apps/configuration/utils/resource-utils.test.tsx

Test Suites: 15 passed, 15 total
Tests:       74 passed, 74 total
Snapshots:   9 passed, 9 total
Time:        16.287s
Ran all test suites matching /public\/apps\/configuration\/utils/i.
✨  Done in 21.48s.
```

Folder-level coverage:

File                                                 | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
-----------------------------------------------------|---------|----------|---------|---------|-------------------
public/apps/configuration/utils                     |   65.32 |    81.68 |   59.02 |   63.91 |                      
  action-groups-utils.tsx                            |   14.29 |    83.33 |   16.67 |    9.09 | 31-32,49-98          
  array-state-utils.tsx                              |    4.17 |        0 |       0 |    4.17 | 20-43,73-120         
  audit-logging-utils.tsx                            |       0 |      100 |       0 |       0 | 21-28                
  auth-view-utils.tsx                                |       0 |      100 |       0 |       0 | 20-21                
  combo-box-utils.tsx                                |      75 |      100 |   66.67 |      75 | 61                   
  context-menu.tsx                                   |   88.89 |      100 |      75 |    87.5 | 40                   
  delete-confirm-modal-utils.tsx                     |   72.73 |       75 |      50 |   77.78 | 35-36                
  display-utils.tsx                                  |   78.95 |    68.42 |   83.33 |   78.95 | 82-96                
  form-row.tsx                                       |     100 |      100 |     100 |     100 |                      
  index-permission-utils.tsx                         |     100 |      100 |     100 |     100 |                      
  internal-user-detail-utils.tsx                     |       0 |      100 |       0 |       0 | 21-29                
  internal-user-list-utils.tsx                       |      25 |       50 |   33.33 |   14.29 | 34-49                
  loading-spinner-utils.tsx                          |     100 |      100 |     100 |     100 |                      
  name-row.tsx                                       |     100 |      100 |     100 |     100 |                      
  panel-with-header.tsx                              |     100 |      100 |     100 |     100 |                      
  password-edit-panel.tsx                            |     100 |      100 |     100 |     100 |                      
  request-utils.ts                                   |       0 |        0 |       0 |       0 | 29-51                
  resource-utils.tsx                                 |     100 |      100 |     100 |     100 |                      
  resource-validation-util.tsx                       |     100 |      100 |     100 |     100 |                      
  role-detail-utils.tsx                              |       0 |      100 |       0 |       0 | 21-25                
  role-list-utils.tsx                                |   44.44 |       70 |   57.14 |    37.5 | 95-108               
  role-mapping-utils.tsx                             |   71.43 |      100 |      60 |      60 | 33,55                
  storage-utils.tsx                                  |      30 |        0 |      50 |      30 | 24-28,42-44          
  tenant-utils.tsx                                   |   84.44 |    95.24 |   64.71 |   82.93 | 50-54,81-101         
  toast-utils.tsx                                    |     100 |      100 |     100 |     100 |                      
  url-builder.tsx                                    |     100 |      100 |     100 |     100 |               

All before:

File                                                 | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
-----------------------------------------------------|---------|----------|---------|---------|-------------------
All files                                            |   62.41 |    57.33 |   48.57 |   63.04 |                      

All after:

File                                                 | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
-----------------------------------------------------|---------|----------|---------|---------|-------------------
All files                                            |   65.99 |    63.72 |   53.17 |   66.55 |                     

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: public/apps/configuration/utils/__snapshots__/panel-with-header.test.tsx.snap [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      Header
+      <i>
+         - optional
+      </i>
+      <span
+        className="panel-header-count"
+      >
+         (
+        0
+        )
```
Comment (tianleh) [inline public/apps/configuration/utils/__snapshots__/panel-with-header.test.tsx.snap:18]: This style looks a bit strange. Can you fix this?

File: public/apps/configuration/utils/action-groups-utils.tsx [hunk 2]
```diff
@@ -32,7 +32,7 @@ export async function fetchActionGroups(http: HttpStart): Promise<DataObject<Act
   return actiongroups.data;
 }
 
-function tranformActionGroupsToListingFormat(
+export function tranformActionGroupsToListingFormat(
```
Comment (tianleh) [inline public/apps/configuration/utils/action-groups-utils.tsx:35]: Typo: tranform -> transform

File: public/apps/configuration/utils/action-groups-utils.test.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import { tranformActionGroupsToListingFormat, mergeAllPermissions } from './action-groups-utils';
```
Comment (tianleh) [inline public/apps/configuration/utils/action-groups-utils.test.tsx:16]: Same typo here.

**Review by tianleh — COMMENTED** [review]
Mostly just minor comments about style and typo.

One comment about the structure. Since we are adding many test files, if we want to go through the non test code, it is a bit annoying since they are mixed with test files. I would suggest moving all such tests to a "test" dir under utils to keep the structure clean. (similar to your "__snapshots__" dir)

File: public/apps/configuration/utils/__snapshots__/panel-with-header.test.tsx.snap (hunk 1 above)
Comment (bandinib-amzn) [reply public/apps/configuration/utils/__snapshots__/panel-with-header.test.tsx.snap:18]: Actually it is correct style. I checked again. We need space before first parenthesis. It looks good in UI.

File: public/apps/configuration/utils/action-groups-utils.tsx (hunk 2 above)
Comment (bandinib-amzn) [reply public/apps/configuration/utils/action-groups-utils.tsx]: Fixed.

File: public/apps/configuration/utils/action-groups-utils.test.tsx (hunk 3 above)
Comment (bandinib-amzn) [reply public/apps/configuration/utils/action-groups-utils.test.tsx]: Fixed.

**Comment by bandinib-amzn** [comment]
> Mostly just minor comments about style and typo.
> 
> One comment about the structure. Since we are adding many test files, if we want to go through the non test code, it is a bit annoying since they are mixed with test files. I would suggest moving all such tests to a "test" dir under utils to keep the structure clean. (similar to your "**snapshots**" dir)

Yes, I agree with you. I will raise new PR for better organization.

**Review by yuxitang-amzn — APPROVED** [review]

**Pull request merged**